Popups feature branch

ClintonYounge opened the pull request:
Add all code necessary for a responsive popup window displaying:

1.  Name, description, featured image, technologies, link to live version, link to source.

NB: All html is created dynamically when the page loads using JavaScript.

ClintonYounge committed:
Add popup window javascript

edgardsoza committed:
add popups css

ClintonYounge committed:
Update css and js

Configured css and js to make popups responsive and more positioned

ClintonYounge committed:
Fix linter errors

ClintonYounge committed:
Fix linter errors

ClintonYounge committed:
Fix linter error

youmari commented on index.html:
```diff
@@ -8,13 +8,16 @@
+  <script src="./popup/popups.js" defer></script>
 </head>
 
 <body>
```
- [ ] you should remove the entire projects section and create it dynamically with javascript in order to complete the requirements

ClintonYounge replied:
@youmari Please explain.

youmari commented on popup/popups.js:
```diff
@@ -0,0 +1,287 @@
+    counter1Img.style.marginRight = '5px';
+
+    const counter2Img = document.createElement('img');
+    counter2Img.src = projects[i].counter2;
```
- [ ] it would be better to stick to the design in order to complete the requirements here are improvements you need to fix 

- [ ] please make sure to use the right spacing like margin padding in order to have some space between elements like in the Figma design 

- [ ] please make sure to add font-weight, font-family font-size and the right color for the text 

- [ ] please make sure that the popup is responsive on all mobile screens 
- Yours 
<img width="292" alt="Screenshot 2023-02-08 at 16 35 24" src="https://user-images.githubusercontent.com/61790587/217578708-1d876619-4fca-4920-8202-b991f7c0fd16.png">

- Figma design 
<img width="230" alt="Screenshot 2023-02-08 at 16 44 52" src="https://user-images.githubusercontent.com/61790587/217585722-4dfdf2f3-6210-4f12-860f-69c437db5153.png">

youmari commented on popup/popups.js:
```diff
@@ -0,0 +1,287 @@
+
+    const counter2Img = document.createElement('img');
+    counter2Img.src = projects[i].counter2;
+    counter2Img.style.marginLeft = '5px';
+    counter2Img.style.marginRight = '5px';
+
+    const positions = document.createElement('div');
+    positions.classList.add('positionsPty');
+
+    positions.append(projects[i].Client, counter1Img, projects[i].Role, counter2Img,
+      projects[i].year);
+
+    card.appendChild(positions);
+
+    const snapshot = document.createElement('img');
+    snapshot.classList.add('snapImages');
+    snapshot.src = projects[i].snapshot;
+    snapshot.style.width = '100%';
+    snapshot.style.height = '300px';
+
+    card.append(snapshot);
+    const description = document.createElement('p');
+    description.textContent = projects[i].description;
+
+    card.append(description);
+    description.style.width = '600px';
+
```
- [ ] I'm unable to see the right side of the pop up please make sure to make it responsive so that it can fit on every screen
<img width="334" alt="Screenshot 2023-02-08 at 17 13 04" src="https://user-images.githubusercontent.com/61790587/217586436-3af1074a-ae07-496e-9dac-662ed06cc5b8.png">

youmari replied:
- [ ] you forgot to add the gray line as in the Figma design please make sure to add it  so that you'll match the design 
<img width="546" alt="Screenshot 2023-02-08 at 17 14 05" src="https://user-images.githubusercontent.com/61790587/217586894-4eed6aec-d6fc-4863-b65b-ebd324937ee2.png">

youmari replied:
- [ ] it would be better to stick to the design in order to complete the requirements here are improvements you need to fix 

- [ ] please make sure to use the right spacing like margin padding in order to have some space between elements like in the Figma design 

- [ ] please make sure to add font-weight, font-family font-size and the right color for the text 

- [ ] please make sure that the popup is responsive on  desktop version
- Yours 
<img width="1433" alt="Screenshot 2023-02-08 at 16 34 29" src="https://user-images.githubusercontent.com/61790587/217587060-bf353bc4-e216-44c8-8fbe-d3c32870a922.png">

- Figma design 
<img width="533" alt="Screenshot 2023-02-08 at 17 16 05" src="https://user-images.githubusercontent.com/61790587/217587197-a0f64ea3-7162-4e43-bd59-1b345414bbcb.png">

youmari reviewed (changes requested):
Hi @ClintonYounge , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ClintonYounge committed:
Update to match template

ClintonYounge committed:
Fix linter  error

ClintonYounge committed:
Add margin to live version button

franclobo commented on popup/popupsPty.css:
```diff
@@ -0,0 +1,207 @@
+  cursor: pointer;
+}
+
+.snapImages {
+  width: 100%;
+  height: 330px;
+}
```
- [ ] You are doing well implementing the required changes 👍🏽  but I can see that the pop-up window doesn't look like the Figma design 🧐, it is because you are trying to show all the windows in the screen size, don't you? but the modal window is bigger than the screen, so please I think we could use `overflow-y: scroll` to allow the modal window to move vertically, kindly do the same in the mobile version because I cannot move to see all the modal for a better user interface.
- [ ] Additionally add the grey opacity margin around the modal window, in the desktop version there is no margin at the bottom and in the mobile version there are no margins on both sides to look similar to the Figma design.

| Figma design | Your design |
|---|---|
| ![image](https://user-images.githubusercontent.com/58642949/217711912-b912f0fd-3404-46a7-a839-155b06eb7117.png) | ![image](https://user-images.githubusercontent.com/58642949/217711822-eac34d16-c749-4f54-b66c-e266ff5f31e3.png) |
| ![image](https://user-images.githubusercontent.com/58642949/217712010-be01f6c3-8456-481f-a279-d44c832bc464.png) | ![image](https://user-images.githubusercontent.com/58642949/217711675-33007d1f-a7a5-4078-b54b-8eeae82bda15.png) |

ClintonYounge replied:
Okay

franclobo reviewed (changes requested):
Hi @ClintonYounge, 

# Required Changes ♻️

![so-close-jim-halpert](https://user-images.githubusercontent.com/58642949/217712122-21127e71-411d-40bd-83c3-51aac76f7982.gif)

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Correct Github flow ✔️ 
- Dynamic and responsive web page ✔️ 
 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ClintonYounge committed:
Update to match template

Add responsive design to match modal.

Add additional screen in between to enhance responsiveness.

ClintonYounge committed:
Increase mobile modal opacity

ClintonYounge committed:
Fix linter errors

ClintonYounge committed:
Fix linter errors

Remove white spaces from css
Place function in root for js

ClintonYounge committed:
Fix linter error

ClintonYounge committed:
Fix spacing linter

Favourezeugwa commented on popup/popups.js:
```diff
@@ -0,0 +1,211 @@
+    description: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic. typesetting, remaining essent.",
+
+    badges: ['html', 'css', 'javascript'],
+    liveBtn: 'See live',
+    sourceBtn: 'See Source',
```
[OPTIONAL] 👍🏽 
- Kindly ensure that your `See Live` and `See Sourece` buttons are active links. You can connect them to their individual links using `href`. 🤗

ClintonYounge replied:
@Favourezeugwa Thank you

Favourezeugwa reviewed (approved):
## STATUS: APPROVED 🥇 🏆 🍹 

Hi @ClintonYounge ,

I am Favour 👩🏽 , your reviewer for this PR.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Yayyy! 💃🏽, Congratulations! 🎉 🎉🎉

![happy](https://user-images.githubusercontent.com/99246359/202491301-53f8c6a2-8ff4-4337-9b6b-9c75c8bdf345.gif)

## Highlights 💯 
- No Linter errors ✔️
- Good commit messages ✔️
- Professional Readme file ✔️
- Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
